### PR TITLE
aezeed: add tests and README

### PR DESCRIPTION
**Pull request by guggero**
I added the text that @Roasbeef wrote in his PR #773 as a README in the `aezeed` package.

Also I added some tests and test vectors, removing one of the TODOs that was in the test.
Additionally, the test file had a typo in it.

### CODE REVIEW

File: aezeed/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+been added, on my laptop I get about 100ms per attempt):
+
+```bash
+⛰ go test -run=XXX -bench=.
+
+goos: darwin
+goarch: amd64
+pkg: github.com/lightningnetwork/lnd/aezeed
+BenchmarkToMnenonic-4     	      10	 102287840 ns/op
+BenchmarkFromMnenonic-4   	      10	 105874973 ns/op
```
Comment (cfromknecht) [inline aezeed/README.md]: While we're at it, let's update these to include the memory allocations as well. Feel free to use your own if you like, here's an output OMM:
```
goos: darwin
goarch: amd64
pkg: github.com/lightningnetwork/lnd/aezeed
BenchmarkTomnemonic-8                 20          87403684 ns/op        33559728 B/op         36 allocs/op
BenchmarkToCipherSeed-8               20          93247937 ns/op        35237697 B/op         39 allocs/op
PASS
ok      github.com/lightningnetwork/lnd/aezeed  6.388s
```

File: aezeed/cipherseed_test.go [hunk 2]
```diff
@@ -34,10 +76,37 @@ func assertCipherSeedEqual(t *testing.T, cipherSeed *CipherSeed,
...
 }
 
+// TestAezeedVersion0TestVectors tests some fixed test vector values against
+// the expected mnemonic words
 func TestAezeedVersion0TestVectors(t *testing.T) {
 	t.Parallel()
 
-	// TODO(roasbeef):
+	for _, v := range version0TestVectors {
+
```
Comment (cfromknecht) [inline aezeed/cipherseed_test.go]: nit: extra newline

File: aezeed/cipherseed_test.go [hunk 3]
```diff
@@ -8,13 +8,55 @@ import (
...
+				"feature", "tissue", "couch", "old", "math", "inform",
+				"success", "suggest", "drink", "motion", "know", "royal",
+			},
+		},
+		{
+			version:  0,
+			time:     time.Unix(1521799345, 0), // 03/23/2018 @ 10:02am (UTC)
+			entropy:  testEntropy,
+			salt:     testSalt,
+			password: []byte("!very_safe_55345_password*"),
```
Comment (cfromknecht) [inline aezeed/cipherseed_test.go]: 👍

File: aezeed/cipherseed_test.go [hunk 4]
```diff
@@ -502,6 +571,45 @@ func TestDecipherIncorrectMnemonic(t *testing.T) {
 
 }
 
+// TestBirthdayCalculationGenesisDate tests that if the bitcoin genesis block time
+// is used, the birthday value of the seed is 0
+func TestBirthdayCalculationGenesisDate(t *testing.T) {
```
Comment (cfromknecht) [inline aezeed/cipherseed_test.go]: It seems these two tests can be collapsed into the test vectors. Both of the vectors are already populated with the timestamps used in these tests, we just need to add an `expectedBirthday` field to the vectors and assert the condition.

File: aezeed/cipherseed_test.go [hunk 5]
```diff
@@ -502,6 +571,45 @@ func TestDecipherIncorrectMnemonic(t *testing.T) {
...
+	// assert out test condition
+	if cipherSeed.Birthday != expectedBirthday {
+		t.Fatalf("mismatched birthday: expected %v, got %v",
+			expectedBirthday, cipherSeed.Birthday)
+	}
+}
+
+// TestBirthdayCalculationSpecificDate tests that if the timestamp 1521799345
+// (03/23/2018 @ 10:02am (UTC)) is used, the birthday value of the seed is 3365
+func TestBirthdayCalculationSpecificDate(t *testing.T) {
```
Comment (cfromknecht) [inline aezeed/cipherseed_test.go]: See comment above

File: aezeed/cipherseed_test.go [hunk 6]
```diff
@@ -8,13 +8,55 @@ import (
 	"time"
 )
 
+// TestVector defines the values that are used to create a fully initialized
+// aezeed mnemonic
```
Comment (cfromknecht) [inline aezeed/cipherseed_test.go]: nit: all comments should preferably be complete sentences :)

**Comment by guggero** [comment]
Thank you for your feedback!
I have collapsed the two birthday calculation tests into the test vector tests as you suggested and fixed the other requests too.

One question I couldn't answer myself by reading the contribution guide: Should I edit/amend my previous commits or add new commits for the changes? I did the latter for the moment but I can change that if requested.

I'm more than happy to continue writing tests! I'll have a look at other packages too.
Unfortunately I don't understand what should be tested to fix the last TODO of the `cipherseed_test.go`:

```
// TODO(roasbeef): add test failure checksum fail is modified, new error
```
There is already a test that validates the checksum. What new error should be added? @Roasbeef any hint?

**Comment by cfromknecht** [comment]
Thanks @guggero! Very happy with the changes here, they LGTM. 🎉

> What new error should be added?
I agree it does sound like that test has already been added, only @Roasbeef can unlock the riddle lol

As far as commit msgs pushing is fine, just so long as they get squashed down before merging. I think we could squash into two commits, one for the tests and one for docs. Once that's done I can approve :)

**Comment by guggero** [comment]
I squashed the commits into two as you requested, @cfromknecht.

**Pull request merged**